Allow insecure mode for HTTPS services

mithun opened the pull request:
Fixes #1

ionelmc commented:
Pasing the options around instead of the positional args is a good change, however i think we should have a separate proto (eg https+insecure://) instead of adding yet another cmdline flag.

mithun commented:
@ionelmc I like the idea of `https+insecure//` proto. However, that might not always be possible in cases where the URL is stored somewhere else (for e.g., an environment variable). I chose the `-k | --insecure` CLI option to match the `curl` option used for the same purpose.

mithun commented:
>What if we don't have a -k shorthand (only --insecure) and also support https+insecure? I can concede to that

@ionelmc That sounds good to me. I will update the PR with those changes sometime this week.

mithun commented:
@ionelmc i've updated the PR to:

* Remove the `-k` shorthand
* Added support for a `https+insecure` proto which behaves identical to `https` with `--insecure`

ionelmc commented:
Looks pretty good. 

One last question: worth making a note of what pythons don't support insecure checks somewhere (like readme)?

mithun commented:
@ionelmc 

>worth making a note of what pythons don't support insecure checks somewhere 

Yes. I've added it to the README

ionelmc commented on src/holdup/cli.py:
```diff
@@ -84,14 +79,52 @@ def __str__(self):
+            proto = 'https'
+        self.url = '{}://{}'.format(proto, urn)
+
+    def can_create_default_context(self):
```
How come the previous feature check was replaced with this version check? Imo this is looking for trouble (eg, pypy doesn't always have same feature parity on version_info).

mithun replied:
Though supported, the `getargspec(urlopen).args` in python3 does not contain `context`. That is because of a `*` in `urllib.request.urlopen`'s argspec [1]. Whereas, `urllib2.urlopen` does not have that[2]. In my testing, the current logic only worked for python2 and not for python3.

Here's an example snippet that shows the difference in behavior:

```python
try:
    from inspect import getfullargspec as getargspec
except ImportError:
    from inspect import getargspec

try:
    from urllib.request import urlopen
except ImportError:
    from urllib2 import urlopen

import pprint


pprint.pprint(getargspec(urlopen).args)
```

Running the above snippet does this on my machine:

```shell
# Python2
$ python ./test.py 
['url', 'data', 'timeout', 'cafile', 'capath', 'cadefault', 'context']

# Python3
$ python3 ./test.py 
['url', 'data', 'timeout']
```

[1] https://github.com/python/cpython/blob/3.6/Lib/urllib/request.py#L141
[2] https://github.com/python/cpython/blob/2.7/Lib/urllib2.py#L131

ionelmc replied:
Right, cause on py3 it's in `kwonlyargs`, not `args`.

mithun replied:
Ah. I did not know that. What is your recommendation on how this should be handled?

mithun replied:
@ionelmc i've switched the logic back to using `inspect` instead of version numbers. Please review.

ionelmc commented:
Very good. I'll remove the py33 from the tests before releasing.

mithun commented:
ping @ionelmc 🔔

ionelmc commented:
1.7.0 is released.
